File: 347-TopKFrequentElements/347-TopKFrequentElements.java
// Last updated: 8/2/2025, 4:53:42 PM
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> counts = new HashMap<>();
        for (int num : nums) {
            counts.put(num, counts.getOrDefault(num, 0) + 1);
        }

        List<int[]> sortedCounts = new ArrayList<>();
        for (int key : counts.keySet()) {
            sortedCounts.add(new int[] {counts.get(key), key});
        }
        sortedCounts.sort((a, b) -> Integer.compare(b[0], a[0]));

        int[] res = new int[k];

        for (int i = 0; i < k; i++) {
            res[i] = sortedCounts.get(i)[1];
        }
        return res;
    }
}

File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 8/2/2025, 4:53:40 PM
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        temp = []
        res = []
        dic = {}
        for num in nums:
            if num in dic:
                dic[num] += 1
            else:
                dic[num] = 1

        for key in dic:
            temp.append([dic[key], key])

        temp.sort(reverse=True)
        
        for i in range(k):
            res.append(temp[i][1])
        
        return res
                